src/Symfony/Component/Validator/Context/ExecutionContext.php

```diff
@@ -139,10 +139,10 @@ public function setConstraint(Constraint $constraint): void
         $this->constraint = $constraint;
     }
 
-    public function addViolation(string $message, array $parameters = []): void
+    public function addViolation(string $message, array $parameters = [], string|false|null $translationDomain = null): void
     {
         $this->violations->add(new ConstraintViolation(
-            $this->translator->trans($message, $parameters, $this->translationDomain),
+            false === $translationDomain ? strtr($message, $parameters) : $this->translator->trans($message, $parameters, $translationDomain ?? $this->translationDomain),
```

Comment:
```suggestion
            false === ($translationDomain ?? $this->translationDomain) ? strtr($message, $parameters) : $this->translator->trans($message, $parameters, $translationDomain ?? $this->translationDomain),
```

Reply:
but do we realy need to add this argument to addViolation? Can't we rely on disableTranslator only instead?
my concern is that this brings knowledge about the translation domain on execution context, while it doesn't know anything about it currently - at the API level.

Reply:
Mmm, I might be wrong, ConstraintViolationBuilderInterface does know about the translation domain, and this execution context returns it, so it also knows about this, transitively.

Reply:
It's not needed indeed, we could rely on the `disableTranslation` method only, in the same way we can rely on the `setTranslation` method only to use another translation domain.

I found more natural and flexible to add it to the method signature, because there was already the parameters as argument, even if a `setParameters` method existed in the `ConstraintViolationBuilderInterface`.

People prefer writing
```
$context->addViolation('Callback message', ['{{ value }}' => 'foobar']);
```
rather than
```
$context->buildViolation('Callback message')->setParameters(['{{ value }}' => 'foobar'])->addViolation();
```
so they could prefer
```
$context->addViolation('Callback message', ['{{ value }}' => 'foobar'], 'custom');
```
rather than
```
$context->buildViolation('Callback message')->setTranslationDomain('custom')->addViolation();
```

But if you prefer I can remove it ; as you wish.

Reply:
To me, the difference is that passing parameters for your message is a common use case. Changing the translation domain for a given violation is a niche use case.

Reply:
Yes, it makes sens ; I remove the param from addViolation